eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/session/send/SessionSender.java

```diff
@@ -139,6 +139,7 @@ public EventMeshTcpSendResult send(Header header, CloudEvent event, SendCallback
                         Objects.requireNonNull(session.getClientGroupWrapper().get())
                             .send(upStreamMsgContext, sendCallback);
                     } finally {
+                        session.getSender().getFailMsgCount().incrementAndGet();
```

Comment:
why you add failMsgCount at here?

Reply:
+1

Reply:
> why you add failMsgCount at here?

In #4103, in fixes section, it was mentioned to add this line at line 142

Reply:
@dipankr what needs to be modified is here:
eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/session/send/UpStreamMsgContext.java line 142

Reply:
updated the PR, moved the line to correct class